usaspending_api/etl/management/commands/update_file_c_linkages_in_delta.py

```diff
@@ -0,0 +1,159 @@
+
+from usaspending_api.awards.delta_models import c_to_d_linkage_view_sql_strings, c_to_d_linkage_drop_view_sql_strings
+from usaspending_api.common.helpers.spark_helpers import (
+    configure_spark_session,
+    get_active_spark_session,
+    get_jvm_logger,
+)
+from usaspending_api.config import CONFIG
+
+
+class Command(BaseCommand):
+
+    help = (
+        "This command links File C (financial_accounts_by_awards) records to File D (awards) records.",
+        "It creates a new copy of the `financial_accounts_by_awards` table in the `int` schema based",
+        "on the copy in the `raw` schema and applies the updates there. These records are linked to",
+        "the `awards` table in the `int` schema",
+    )
+
+    spark: SparkSession
+
+    def add_arguments(self, parser):
+        parser.add_argument(
+            "--no-clone",
+            action="store_true",
+            required=False,
+            help="Whether to create a full new `int` table instead of a clone.",
+        )
+
+        parser.add_argument(
+            "--spark-s3-bucket",
+            type=str,
+            required=False,
+            default=CONFIG.SPARK_S3_BUCKET,
+            help="The destination bucket in S3 to write the data",
+        )
+
+    def get_unlinked_count(self, schema):
+        unlinked_sql = f"""
+        SELECT
+            count(*)
+        FROM
+            {schema}.financial_accounts_by_awards
+        WHERE
+            award_id IS NULL;
+        """
+
+        results = self.spark.sql(unlinked_sql)
+        return results.collect()[0][0]
+
+    def handle(self, *args, **options):
+
+        extra_conf = {
+            # Config for Delta Lake tables and SQL. Need these to keep Dela table metadata in the metastore
+            "spark.sql.extensions": "io.delta.sql.DeltaSparkSessionExtension",
+            "spark.sql.catalog.spark_catalog": "org.apache.spark.sql.delta.catalog.DeltaCatalog",
+            # See comment below about old date and time values cannot parsed without these
+            "spark.sql.legacy.parquet.datetimeRebaseModeInWrite": "LEGACY",  # for dates at/before 1900
+            "spark.sql.legacy.parquet.int96RebaseModeInWrite": "LEGACY",  # for timestamps at/before 1900
+            "spark.sql.jsonGenerator.ignoreNullFields": "false",  # keep nulls in our json
+        }
+
+        no_clone = options["no_clone"]
+        spark_s3_bucket = options["spark_s3_bucket"]
+
+        self.spark = get_active_spark_session()
+        spark_created_by_command = False
+        if not self.spark:
+            spark_created_by_command = True
+            self.spark = configure_spark_session(**extra_conf, spark_context=self.spark)  # type: SparkSession
+
+        # Setup Logger
+        logger = get_jvm_logger(self.spark, __name__)
+
+        # Setup int table. Creates a shallow clone of the `raw` FABA table in the `int` schema.
+        # If the --no-clone option is provided a full table is created instead.
```

Comment:
Maybe also clarify under --no-clone that it is a _managed_ table, meaning the data is not stored in a specific S3 bucket
- Do we know where it gets stored when managed?
- i.e. do we at least know that it's not stored on the filesystem of the driver or the cluster nodes, because that data would disappear if so when the cluster is terminated.

Reply:
With it now not being managed, you may need to or want to purge the data from the bucket after the table is deleted with the vacuum with retention set to 0 hours. I think the default vacuum is 30 days. So we'd have 30 copies of the data in the S3 bucket before it starts to clean up. 
Actually, does that matter or apply at all with it being a shallow clone? Maybe it just stores _new_ files that are different than the cloned destination? That may be okay to not forcibly vacuum.

Reply:
Also, in the `--no-clone` scenario, is the `TABLE_SPEC` setup to create the table with the same S3 bucket and path?

```
        # Setup int table. Creates a shallow clone of the `raw` FABA table in the `int` schema.
        # If the --no-clone option is provided a full table is created instead.
        if no_clone:
            self.spark.sql("DROP TABLE IF EXISTS int.financial_accounts_by_awards")
            call_command(
                "create_delta_table",
                "--destination-table",
                "financial_accounts_by_awards",
                "--spark-s3-bucket",
                spark_s3_bucket,
                "--alt-db",
                "int",
            )
```

Reply:
Resolving. Summary:
- vacuuming needs to be addressed more holistically as this affects all of our tables (and s3 paths)
- creat_delta_table uses s3: Bucket is passed in as a param. Derive path from the schema and table name in the table spec